pkg/resources/resources_manager_test.go

```diff
@@ -485,4 +486,44 @@ var _ = Describe("ResourcesManger", func() {
 			fakeResourceServer.AssertExpectations(testCallsAssertionReporter)
 		})
 	})
+	Context("PeriodicUpdate", func() {
+		It("Update resources for resource manager", func() {
+			fs := &utils.FakeFilesystem{
+				Dirs: []string{
+					"sys/bus/pci/devices/0000:02:00.0",
+				},
+				Files: map[string][]byte{
+					"sys/bus/pci/devices/0000:02:00.0/modalias": []byte(
+						"pci:v000015B3d00001017sv000015B3sd00000001bc02sc00i00"),
+				},
+			}
+			defer fs.Use()()
+			os.Setenv("GHW_CHROOT", fs.RootDir)
+			defer os.Unsetenv("GHW_CHROOT")
+
+			fakeResourceServer := mocks.ResourceServer{}
+			fakeResourceServer.On("UpdateDevices", mock.Anything).Return()
+
+			nLink := &mocks.NetlinkManager{}
+			link := &FakeLink{netlink.LinkAttrs{EncapType: "ether"}}
+			nLink.On("LinkByName", mock.Anything).Return(link, nil)
+
+			rds := &mocks.RdmaDeviceSpec{}
+			rds.On("Get", mock.Anything).Return([]*pluginapi.DeviceSpec{})
+			rds.On("VerifyRdmaSpec", mock.Anything).Return(nil)
+
+			configList := []*types.UserConfig{{Selectors: types.Selectors{Vendors: []string{"Vendors"}}}}
+			rm := &resourceManager{
+				configList:             configList,
+				resourceServers:        []types.ResourceServer{&fakeResourceServer},
+				netlinkManager:         nLink,
+				rds:                    rds,
+				PeriodicUpdateInterval: 1 * time.Millisecond,
+			}
+
+			rm.PeriodicUpdate()
+			time.Sleep(2 * time.Second)
+			Expect(len(rm.deviceList)).To(Equal(1))
```

Comment:
can you also ensure that UpdateDevices was called on fakeResourceServer ?

Reply:
added